fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneMailExampleTest.java

```diff
@@ -0,0 +1,257 @@
+                entries.next().getKey();
+            }
+            commit(context);
+        }
+
+    }
+
+
+    @Test
+    public void testWritePerformance() throws Exception {
+        loadData();
+        LOGGER.info("Write Execution");
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, "Email", MAIL_EXAMPLE_INDEX);
+            loadRecords(DATA_RESOURCE_DIRECTORY, 1, 1000);
+            commit(context);
+        }
+    }
+
+    @Test
+    public void testUpdatePerformance() throws Exception {
+        loadData();
+        LOGGER.info("Update Execution");
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, "Email", MAIL_EXAMPLE_INDEX);
+            loadRecords(DATA_RESOURCE_DIRECTORY, 1, 1);
+            commit(context);
+        }
+    }
+
+    private void loadData() throws Exception {
+        for (int i = 0; i < RECORD_COUNT_NUM_BATCHES; i++) {
+            try (FDBRecordContext context = openContext()) {
+                rebuildIndexMetaData(context, "Email", MAIL_EXAMPLE_INDEX);
+                loadRecords(DATA_RESOURCE_DIRECTORY, RECORD_COUNT_BATCH_SIZE, i * RECORD_COUNT_BATCH_SIZE);
+                commit(context);
+            }
+        }
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, useRewritePlanner);
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    /**
+     * Load the records for emails from the EML files.
+     * @param dataResourceDirectory the directory name to load EML fields from
+     * @param recordCountToLoad count of records to load
+     * @throws Exception exception
+     */
+    void loadRecords(String dataResourceDirectory, int recordCountToLoad, int offset) throws Exception {
+        final File file = new File(dataResourceDirectory);
+        final Properties props = System.getProperties();
+        int index = 0;
+        int mailBoxIndex = 0; // this is ignored if Gaussian distribution is used
+        Random random = new Random();
+
+        long totalWriteTime = 0L;
+        // Write mbox records
+
+        List<String> mboxIds = new ArrayList<>();
+        for (int i = 1; i <= (USE_GAUSSIAN_DISTRIBUTION ? MAIL_BOX_COUNT : MAIL_BOX_DISTRIBUTION.length); i++) {
+            mboxIds.add("box" + i);
+        }
+        for (File emlFile : file.listFiles()) {
+            int mboxId = USE_GAUSSIAN_DISTRIBUTION ? (int)(random.nextGaussian() * (MAIL_BOX_COUNT / 6) + (MAIL_BOX_COUNT / 2 + 1)) : mailBoxIndex + 1;
+            mboxId = Math.min(mboxId, MAIL_BOX_COUNT);
+            mboxId = Math.max(mboxId, 1);
+            Session session = Session.getDefaultInstance(props, null);
+            InputStream source = new FileInputStream(emlFile);
```

Comment:
Not sure if that's a concern, but this looks to start each batch with the first file (and only change the record ID)

Reply:
Yes, that is understood.